pager/display.c

```diff
@@ -178,7 +178,7 @@ static void resolve_color(struct MuttWindow *win, struct Line *lines, int line_n
   }
 
   color = def_color;
-  if ((flags & MUTT_SHOWCOLOR) && lines[m].syntax_arr_size)
+  if ((flags & MUTT_SHOWCOLOR) && (lines[m].syntax_arr_size > 0))
```

Comment:
I read into `bsearch()` a bit because I was suprised that it would fail with an empty array. And if I understand [the standard](https://pubs.opengroup.org/onlinepubs/9699919799/functions/bsearch.html) correctly it shouldn't:
> If the nel argument has the value zero, the comparison function pointed to by compar shall not be called and no match shall be found.

So the problem isn't an empty array but `lines[m].syntax` being `NULL`?

Long story short: while I understand how this prevents the crash wouldn't this be a more "correct" fix?

```suggestion
  if ((flags & MUTT_SHOWCOLOR) && lines[m].syntax)
```

Reply:
Yeah.  Updated.

I don't know why, but the number of elements is initialised to `-1`.
This makes all the `if (num)` checks a bit iffy.

The code needs to die, but refactoring the pager is a huge block of work